bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -936,6 +996,19 @@ void ReplicaImp::startConsensusProcess(PrePrepareMsg *pp, bool isCreatedEarlier)
     sendPartialProof(seqNumInfo);
   }
 }
+PrePrepareMsg *ReplicaImp::getConsensusPPFromDataPP(PrePrepareMsg *pp) {
+  if (pp == nullptr) return pp;
+  // set flag for data pp msg
+  // return another pp message with consensus only flag
+  PrePrepareMsg *consensus_only_pp = pp->createConsensusPPMsg(pp);
+  // store it in the map
+  auto digest = consensus_only_pp->digestOfRequests().toString();
+  auto dataPP = pp->cloneDataPPMsg(pp);
+  // TODO(NK): need to persist this
+  // TODO(NK): check and remove old data messages on stable seqNum
```

Comment:
you can remove the data message once you attach the data to the consensusPP message (removing it also means deleting the pointer)